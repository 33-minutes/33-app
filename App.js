import React, { Component } from 'react';
import { createRootNavigator } from './app/Main';
import environment from './app/Environment';
import { SERVER_URL } from 'react-native-dotenv';

export default class App extends Component {
  state = {
    signedIn: false
  }

  constructor(props) {
    super(props);

    console.log("Starting 33 Minutes, connecting to " + SERVER_URL + ".");
  }

  render() {
    const { signedIn } = this.state;
    
    const screenProps = { 
      relay: {
        environment: environment
      }
    }

    const Layout = createRootNavigator(signedIn, screenProps);

    return (
      <Layout screenProps={ screenProps } />
    );
  }
}
