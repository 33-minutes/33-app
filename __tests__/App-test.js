import React from 'react';
import App from '../App';

import renderer from 'react-test-renderer';

describe('app', () => {
    const app = renderer.create(<App />);

    it('renders app', () => {
        const rendered = app.toJSON();
        expect(rendered).toMatchSnapshot()
    });
})

