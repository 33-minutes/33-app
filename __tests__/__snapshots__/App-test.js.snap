// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders app 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <RCTSafeAreaView
          style={
            Object {
              "backgroundColor": "#f2f2f2",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#f2f2f2",
                  "elevation": 4,
                  "shadowColor": "black",
                  "shadowOffset": Object {
                    "height": 0.5,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 0.5,
                  "zIndex": 1,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#ffeb3b",
                      "bottom": 0,
                      "height": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "overflow": "scroll",
                  }
                }
              >
                <RCTScrollView
                  alwaysBounceHorizontal={false}
                  automaticallyAdjustContentInsets={false}
                  contentContainerStyle={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "flexWrap": "nowrap",
                      },
                      Object {
                        "flex": 1,
                      },
                    ]
                  }
                  horizontal={true}
                  keyboardShouldPersistTaps="handled"
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  overScrollMode="never"
                  scrollEventThrottle={1}
                  scrollsToTop={false}
                  showsHorizontalScrollIndicator={false}
                >
                  <View>
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel="You"
                      accessibilityTraits={
                        Array [
                          "button",
                          "selected",
                        ]
                      }
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "justifyContent": "center",
                              "opacity": 1,
                              "padding": 8,
                              "paddingTop": 8,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 24,
                                  "width": 24,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "height": "100%",
                                  "justifyContent": "center",
                                  "minWidth": 25,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "height": "100%",
                                  "justifyContent": "center",
                                  "minWidth": 25,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                              />
                            </View>
                          </View>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "rgba(0, 0, 255, 1)",
                                "fontSize": 13,
                                "margin": 8,
                                "textAlign": "center",
                              }
                            }
                          >
                            YOU
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      accessibilityComponentType="button"
                      accessibilityLabel="Following"
                      accessibilityTraits="button"
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "flex": 1,
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flex": 1,
                              "justifyContent": "center",
                              "opacity": 0.7,
                              "padding": 8,
                              "paddingTop": 8,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "height": 24,
                                  "width": 24,
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "height": "100%",
                                  "justifyContent": "center",
                                  "minWidth": 25,
                                  "opacity": 0,
                                  "position": "absolute",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "height": "100%",
                                  "justifyContent": "center",
                                  "minWidth": 25,
                                  "opacity": 1,
                                  "position": "absolute",
                                  "width": "100%",
                                }
                              }
                            >
                              <Text
                                accessible={true}
                                allowFontScaling={true}
                                ellipsizeMode="tail"
                              />
                            </View>
                          </View>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "rgba(128, 128, 128, 1)",
                                "fontSize": 13,
                                "margin": 8,
                                "textAlign": "center",
                              }
                            }
                          >
                            FOLLOWING
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <RCTScrollView
                alwaysBounceHorizontal={false}
                automaticallyAdjustContentInsets={false}
                bounces={false}
                contentContainerStyle={
                  Object {
                    "flex": 1,
                  }
                }
                contentOffset={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                directionalLockEnabled={true}
                horizontal={true}
                keyboardDismissMode="on-drag"
                keyboardShouldPersistTaps="always"
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                overScrollMode="never"
                pagingEnabled={true}
                scrollEnabled={true}
                scrollEventThrottle={1}
                scrollsToTop={false}
                showsHorizontalScrollIndicator={false}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View>
                  <View
                    accessibilityElementsHidden={false}
                    importantForAccessibility="auto"
                    style={
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <RCTScrollView>
                          <View>
                            <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                Object {
                                  "padding": 20,
                                }
                              }
                            >
                              Press the record button below to record a meeting.
                            </Text>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                    style={null}
                  />
                </View>
              </RCTScrollView>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 5,
                  },
                ]
              }
            >
              <View
                padding={0}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "padding": 8,
                    },
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 5,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#F7F7F7",
              "borderBottomColor": "#A7A7AA",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, .9)",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  33 Minutes
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
